pkg/backend/diy/stack.go

```diff
@@ -54,6 +54,26 @@ func newStack(ref *diyBackendReference, b *diyBackend) backend.Stack {
 }
 
 func (s *diyStack) Ref() backend.StackReference { return s.ref }
+func (s *diyStack) GetStackFilename(ctx context.Context) (string, bool) {
+	_, path, err := workspace.DetectProjectStackPath(s.Ref().Name().Q())
+	return path, err == nil
+}
+
+func (s *diyStack) Load(ctx context.Context, project *workspace.Project, configFileOverride string,
+) (*workspace.ProjectStack, error) {
+	if configFileOverride != "" {
+		return workspace.LoadProjectStack(project, configFileOverride)
+	}
+	return workspace.DetectProjectStack(s.Ref().Name().Q())
+}
+
+func (s *diyStack) Save(ctx context.Context, projectStack *workspace.ProjectStack, configFileOverride string) error {
+	if configFileOverride != "" {
+		return projectStack.Save(configFileOverride)
+	}
+	return workspace.SaveProjectStack(s.Ref().Name().Q(), projectStack)
+}
+
```

Comment:
I feel a bit off about this. Like the file based load/save should wrap around Stack so that Stack itself would instead have load and save without the configFileOverride, and instead of GetStackFilename have like "HasRemoteConfig() bool"